server/src/main/java/org/elasticsearch/cluster/metadata/ComponentTemplate.java

```diff
@@ -121,7 +142,8 @@ public boolean equals(Object obj) {
         ComponentTemplate other = (ComponentTemplate) obj;
         return Objects.equals(template, other.template)
             && Objects.equals(version, other.version)
-            && Objects.equals(metadata, other.metadata);
+            && Objects.equals(metadata, other.metadata)
+            && deprecated() == other.deprecated();
```

Comment:
I think this should be
```suggestion
            && Objects.equals(deprecated, other.deprecated);
```

Because otherwise, a template with a missing `deprecated` field is equivalent to a template with `deprecated: false`, and it's possible that the service could noop an update when it shouldn't.